scipy/stats/meson.build

```diff
@@ -0,0 +1,157 @@
+fortran_ignore_warnings = [
+  '-Wno-tabs', '-Wno-conversion',
+  '-Wno-argument-mismatch', '-Wno-unused-dummy-argument',
+  '-Wno-maybe-uninitialized', '-Wno-unused-label',
+  '-Wno-unused-variable']
+
+subdir('tests')
+
+statlib_lib = static_library('statlib_lib',
+    ['statlib/ansari.f',
+     'statlib/spearman.f',
+     'statlib/swilk.f'])
+
+statlib_module = custom_target('statlib_module',
+  output : ['statlib-f2pywrappers.f', 'statlibmodule.c'],
+  input : 'statlib.pyf',
+  command : [py3, generate_f2pymod, '@INPUT@', '-o', '@OUTDIR@'],
+)
+
+statlib = py3.extension_module('statlib',
+  [statlib_module],
+  c_args: numpy_nodepr_api,
+  fortran_args: fortran_ignore_warnings,
+  link_with : statlib_lib,
+  dependencies : [py3_dep],
+  include_directories : [inc_np, inc_f2py],
+  install : true,
+  subdir : 'scipy/stats')
+
+# TODO: Add c_args: numpy_nodepr_api 
+_stats = py3.extension_module('_stats',
+  ['_stats.pyx'],
```

Comment:
This actually doesn't work yet because it relies on `special` which we haven't wrapped yet:
```
FAILED: scipy/stats/_stats.cpython-39-x86_64-linux-gnu.so.p/scipy/stats/_stats.pyx.c 
cython --fast-fail -3 -o scipy/stats/_stats.cpython-39-x86_64-linux-gnu.so.p/scipy/stats/_stats.pyx.c /home/rgommers/code/bldscipy/scipy/stats/_stats.pyx

Error compiling Cython file:
------------------------------------------------------------
...
from numpy cimport ndarray, int64_t, float64_t, intp_t

import warnings
import numpy as np
import scipy.stats, scipy.special
cimport scipy.special.cython_special as cs
```
I'll comment it out.

Reply:
Before pushing, I did a `git clean -fdx`/deleted build/install dir and then rebuilt the project, the above error wasn't thrown. Commenting out is fine. We can go for `spatial` and `special` first now.

Reply:
Sometimes the error is hard to spot, because of the parallel build - other output (like warnings) get put below it, and then you have to scroll way up to see the actual problem. This is why I want to silence or fix irrelevant warnings, easier to find what matters.